### PR TITLE
feat: [STO-4293] Add markdown component

### PR DESCRIPTION
**Pull request by mbaguiar**
### What this PR does

1. Adds markdown component

Screenshot 2022-07-18 at 11 56 10

### Why is this needed?

To have a reusable component that renders markdown text, that can be used across different services.

Solves:  
STO-4293

### Checklist:

- [x] I reviewed my own code
- [x] The changes align with the designs I received  
  Or give a reason why this does not apply:
- [x] I have attached screenshot(s), video(s) or gif(s) showing that the solution is working as expected  
  Or give a reason why this does not apply:
- [x] I have updated the task(s) status on Linear
- [x] All new media is optimized (images, gifs, videos)

### Browser support

My code works in the following browsers:

- [x] Firefox
- [x] Chrome
- [x] Safari
- [x] Edge

### CODE REVIEW

File: src/lib/components/markdown/index.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+<Preview>
+  <Markdown>{example}</Markdown>
+</Preview>
+
+| attribute          | unit                                           | description                                      | default value | required |
+| ------------------ | ---------------------------------------------- | ------------------------------------------------ | ------------- | -------- |
+| children           | string                                         | Markdown source to be rendered                   | n/a           | true     |
+| customMDComponents | Record&ltstring, FunctionComponent&ltany&gt&gt | Styling functions for custom tags                | n/a           | false    |
+| className          | string                                         | Styles to be applied to the whole markdown block | ''            | false    |
+| openLinksInNewTab  | boolean                                        | Wether the links should open in a new tab        | false         | false    |
```
Comment (kimkwanka) [inline src/lib/components/markdown/index.stories.mdx:21]:
```suggestion
| openLinksInNewTab  | boolean                                        | Whether the links should open in a new tab        | false         | false    |
```

Comment (mbaguiar) [reply src/lib/components/markdown/index.stories.mdx:21]: 🙈

File: src/lib/components/markdown/example.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Headers
```
Comment (keremciu) [inline src/lib/components/markdown/example.md:1]: this is a good approach but I'm curious about show code functionality of storybook. do we want to see this usages there? if so, now how it looks {example} or example.md content

**Review by keremciu — APPROVED** [review]

File: src/lib/components/markdown/example.md (hunk 2 above)
Comment (mbaguiar) [reply src/lib/components/markdown/example.md:1]: Yeah, using this example.md file, its contents don't show on the `Show code` feature. However, I added raw markdown and rendered markdown to the example, so you can see how everything is used there.

Comment (keremciu) [reply src/lib/components/markdown/example.md:1]: great! thank you for landing this component ✨

**Review by talo242 — APPROVED** [review]
🙌🏼

**Review by kseha1080 — APPROVED** [review]
✌️!

**Pull request merged**